src/Region/RegionHandler.cc

```diff
@@ -1557,214 +1776,336 @@ std::vector<int> RegionHandler::GetProjectedFileIds(int region_id) {
+    const std::string& coordinate, RegionState& region_state, std::shared_ptr<casacore::CoordinateSystem> reference_csys,
+    std::function<void(float)>& progress_callback, casacore::Matrix<float>& profiles, double& increment, bool& cancelled) {
     // Calculate mean spectral profiles for box regions along line with fixed pixel spacing, with progress updates after each profile.
     // Return parameters include the profiles, the increment between the box centers in arcsec, and whether profiles were cancelled.
     // Returns false if profiles cancelled or linear pixel centers are tabular in world coordinates.
-    auto endpoints = region_state.control_points;
-    auto dx_pixels = endpoints[1].x() - endpoints[0].x();
-    auto dy_pixels = endpoints[1].y() - endpoints[0].y();
-    size_t num_regions = sqrt((dx_pixels * dx_pixels) + (dy_pixels * dy_pixels));
+    auto control_points = region_state.control_points;
+    size_t num_lines(control_points.size() - 1);
 
-    // Offset range [-offset, offset] from center, in pixels
-    int num_offsets = lround(float(num_regions - 1) / 2.0);
-    auto center_idx = num_offsets;
-    std::vector<CARTA::Point> box_centers((num_offsets * 2) + 1);
-
-    // Center point of line
-    auto center_x = (endpoints[0].x() + endpoints[1].x()) / 2;
-    auto center_y = (endpoints[0].y() + endpoints[1].y()) / 2;
-
-    // Set center pixel at center index
-    CARTA::Point point;
-    point.set_x(center_x);
-    point.set_y(center_y);
-    box_centers[center_idx] = point;
-
-    // Apply rotation to get next pixel
-    auto rotation = region_state.rotation;
-    float cos_x = cos((rotation + 90.0) * M_PI / 180.0f);
-    float sin_x = sin((rotation + 90.0) * M_PI / 180.0f);
-
-    // Set pixel direction for horizontal line
-    int pixel_dir(1.0);
-    if (dy_pixels == 0.0) {
-        pixel_dir = (dx_pixels < 0 ? -1.0 : 1.0);
-    }
-
-    // Set pixels in pos and neg direction from center out
-    for (int ipixel = 1; ipixel <= num_offsets; ++ipixel) {
-        // Positive offset
-        auto idx = center_idx + ipixel;
-        point.set_x(center_x - (pixel_dir * ipixel * cos_x));
-        point.set_y(center_y - (pixel_dir * ipixel * sin_x));
-        box_centers[idx] = point;
-
-        // Negative offset
-        idx = center_idx - ipixel;
-        point.set_x(center_x + (pixel_dir * ipixel * cos_x));
-        point.set_y(center_y + (pixel_dir * ipixel * sin_x));
-        box_centers[idx] = point;
-    }
+    auto t_start = std::chrono::high_resolution_clock::now();
+    float progress(0.0);
 
-    if (per_z && _stop_pv[file_id]) {
-        cancelled = true;
-        return false;
-    }
+    if (num_lines == 1) {
+        // Get box centers along line; start at line center and add offsets
+        PointXy endpoint0(control_points[0].x(), control_points[0].y());
+        PointXy endpoint1(control_points[1].x(), control_points[1].y());
```

Comment:
Perhaps these could also have more descriptive start / end names? Otherwise this could be a vector rather than two numbered variables. Would that have any effect on performance?

Reply:
Renamed to line_start and line_end. I think it is more readable to keep them separarte, and the endpoints are frequently used separately in the code (e.g. positive offsets vs negative offsets). I also changed to `vector<double>` instead of `PointXy` to make FixedPixelRegionProfiles consistent with FixedAngularRegionProfiles, and because the functions within the struct were not needed.